app/src/main/java/com/manage/ak/moneyreport/MainActivity.java

```diff
@@ -0,0 +1,555 @@
+            for (Sms r : (List<Sms>) i.readObject()) {
+                cashList.add(r);
+            }
+            i.close();
+            f.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        if (!savedData.getString("SPENT", "").equals(""))
+            CASHSPENT = savedData.getString("SPENT", "");
+
+        bankBalance = (TextView) findViewById(R.id.bankBalance);
+        estimateDate = (TextView) findViewById(R.id.estimateDate);
+
+        spentAmount = (TextView) findViewById(R.id.spendAmount);
+        cashBalance = (TextView) findViewById(R.id.cashBalance);
+
+        spentAmount.setText("₹ " + CASHSPENT);
+
+        if (cashList.size() > 0) {
+            cashBalance.setText("In Hand:   ₹ " + cashList.get(0).getMsgBal());
+        } else {
+            cashBalance.setText("In Hand:   ₹ " + "0.0");
+        }
+
+        if (bankList.size() > 0) {
+            bankBalance.setText("₹ " + bankList.get(0).getMsgBal());
+            estimateDate.setText(bankList.get(0).getFormatDate());
+        } else {
+            bankBalance.setText("₹ " + "0.0");
+            estimateDate.setText(" ");
+        }
+
+        CardView bankCard = (CardView) findViewById(R.id.bankCard);
+        bankCard.setOnClickListener(this);
+
+        CardView cashCard = (CardView) findViewById(R.id.cashCard);
+        cashCard.setOnClickListener(this);
+
+        TextView refresh = (TextView) findViewById(R.id.refresh);
+        refresh.setOnClickListener(this);
+
+        TextView addCash = (TextView) findViewById(R.id.addCash);
+        addCash.setOnClickListener(this);
+
+        TextView bankReport = (TextView) findViewById(R.id.bankReport);
+        bankReport.setOnClickListener(this);
+
+        TextView cashReport = (TextView) findViewById(R.id.cashReport);
+        cashReport.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.bankCard:
+                if (bankList.size() > 0) {
+                    Intent bank = new Intent(MainActivity.this, BankTransactions.class);
+                    Bundle b = new Bundle();
+                    b.putSerializable("SMS", (Serializable) bankList);
+                    bank.putExtra("DATA", b);
+                    startActivity(bank);
+                } else {
+                    Toast.makeText(MainActivity.this, "No Bank Transactions to display", Toast.LENGTH_SHORT).show();
+                }
+                break;
+            case R.id.cashCard:
+                Intent cash = new Intent(MainActivity.this, CashTransactions.class);
+                Bundle b = new Bundle();
+                b.putSerializable("CASH", (Serializable) cashList);
+                b.putString("Spent", CASHSPENT);
+                cash.putExtra("DATA", b);
+                startActivityForResult(cash, 1);
+                break;
+            case R.id.refresh:
```

Comment:
There should not be any manual refresh for the user. You should read the messages the first time user enters the app and thereafter listen to any new incoming messages.